### PR TITLE
Dev 2071 remove quarantine from contributions

### PR DESCRIPTION
**Pull request by harsha09**
Please fill out each section even if it's just with "N/A"

#### What's this PR do?
Stop showing flagged contributions in the contributions dashboard.
Removes UI elements (filters, table columns) related to flagged contributions.

#### Why are we doing this? How does it help us?
This is a sub-step involved in quarantine management. Subsequent ticket will create another view to show the quarantine and the user will be able to directly manage Approval/Rejection of flagged contributions.

#### How should this be manually tested? Please include detailed step-by-step instructions.
Login to the app -> Click on Contributions dashboard.
Flagged contributions shouldn't be displayed.
`Flagged` and `Rejected` filters shouldn't be shown.
`Date flagged` and `Auto-resolution Date` shouldn't be shown in table.

Approving/Rejecting a flagged contribution:
--------------------------------------------
Login to Admin panel -> Contributions tab -> Select any flagged contribution -> Select `Accept flagged contributions` in Action dropdown -> Go
Contribution should get displayed in the contributions dashboard

Same with the reject scenario but the contribution will not get displayed in the contributions dashboard.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?
Yes

#### Have automated unit tests been added? If not, why?
Yes

#### How should this change be communicated to end users?
May be using user-manual.

#### Are there any smells or added technical debt to note?
No

#### Has this been documented? If so, where?
Not yet

#### What are the relevant tickets? Add a link to any relevant ones.
[DEV-2071]

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?
NA

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:
[DEV-2250]
[DEV-1619]

[DEV-2071]: https://news-revenue-hub.atlassian.net/browse/DEV-2071?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: spa/src/components/donations/Donations.js [hunk 1]
```diff
@@ -28,6 +28,8 @@ import { StatusCellIcon } from 'components/contributor/contributorDashboard/Cont
 import Filters from 'components/donations/filters/Filters';
 import GenericErrorBoundary from 'components/errors/GenericErrorBoundary';
 import PageTitle from 'elements/PageTitle';
+import { PAYMENT_STATUS } from 'constants';
+import { PAYMENT_STATUS_EXCLUDE_IN_CONTRIBUTIONS } from 'constants';
```
Comment (GuiMend) [inline spa/src/components/donations/Donations.js:32]:
```suggestion
import { PAYMENT_STATUS, PAYMENT_STATUS_EXCLUDE_IN_CONTRIBUTIONS  } from 'constants';
```

File: spa/src/components/donations/filters/StatusFilter.js [hunk 2]
```diff
@@ -7,22 +7,25 @@ import { PAYMENT_STATUS } from 'constants';
...
-              selected={filter.includes(f)}
-              onClick={() => handleFilterChange('status', f)}
-            >
-              <StatusCellIcon status={f} showText size="sm" />
-            </S.StatusBadge>
-          ))}
+          {STATUS_FILTERS.map(
+            (f) =>
+              excludeStatusFilters.indexOf(f) == -1 && (
+                <S.StatusBadge
```
Comment (GuiMend) [inline spa/src/components/donations/filters/StatusFilter.js:19]: I would prefer to use something like the `filter` method:
You can add the `!` to choose if you want to select the subset, or if you want to exclude the subset values.

image
```suggestion
          {STATUS_FILTERS.filter(status => excludeStatusFilters.includes(status)).map(
            (status) => (
                <S.StatusBadge
```

File: spa/src/components/donations/filters/StatusFilter.js [hunk 3]
```diff
@@ -7,22 +7,25 @@ import { PAYMENT_STATUS } from 'constants';
...
+                  key={f}
+                  data-testid={`status-filter-${f}`}
+                  selected={filter.includes(f)}
+                  onClick={() => handleFilterChange('status', f)}
+                >
+                  <StatusCellIcon status={f} showText size="sm" />
+                </S.StatusBadge>
```
Comment (GuiMend) [inline spa/src/components/donations/filters/StatusFilter.js:26]: Also, I didn't see this when I edited this page, but could you rename `f` for `status` or another more descriptive name? 
`f` is really hard to understand what it stands for 
```suggestion
                  key={status}
                  data-testid={`status-filter-${status}`}
                  selected={filter.includes(status)}
                  onClick={() => handleFilterChange('status', status)}
                >
                  <StatusCellIcon status={status} showText size="sm" />
                </S.StatusBadge>
```

**Review by GuiMend — CHANGES_REQUESTED** [review]
I have a few comments regarding the code, but the functionality is working as expected

Flags are not showing 👍 
image

**Review by njharman — APPROVED** [review]
[Only reviewed Python backend]

Code 100% covered.
It doesn't seem that the QA instructions include checking AC 4 or 5. Which seem important to check.

**Review by GuiMend — APPROVED** [review]
Changes made 👍

**Review by x110dc — APPROVED** [review]

**Review by x110dc — APPROVED** [review]

**Pull request merged**